int32 upTime
int32 idleTime
int32 upperTime
int32 lowerTime
